### PR TITLE
Wordpress preview integration

### PR DESCRIPTION
**Pull request by thomasdigby**
Integrated custom draft API to match [WP Tapestry Plugin](https://github.com/shortlist-digital/wp-tapestry-plugin). Still need to write tests.

### CODE REVIEW

File: test/tests/routing.test.js [hunk 1]
```diff
@@ -190,3 +190,53 @@ describe('Handling custom endpoint routes', () => {
...
+
+    it('Preview route without embed, endpoint works', (done) => {
+      request.get(`${uri}/path-without-embed?tapestry_hash=hash&p=10`, (err, res, body) => {
+        expect(body).to.contain(prepareJson(dataPage))
+        done()
+      })
+    })
+
+    it('Preview route with embed, endpoint works', (done) => {
+      request.get(`${uri}/path-with-embed?tapestry_hash=hash&p=10`, (err, res, body) => {
```
Comment (jonsherrard) [inline test/tests/routing.test.js]: Should this be `${uri}/path-with-embed?_embed&tapestry_hash=hash&p=10`?

File: src/server/handle-preview.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  server.route({
+    method: 'GET',
+    path: '/api/preview/v1/{query*}',
+    handler: (req, reply) => {
+
+      const base = `${config.siteUrl}/wp-json/revision/v1`
+      const path = `${req.params.query}${req.url.search}`
+      const remote = `${base}/${path}`
+
+      winston.log('debug', `Server loading API response over HTTP for ${remote}`)
```
Comment (jonsherrard) [inline src/server/handle-preview.js]: I know we didn't settle on a grammar for logging, and that req.url.search will be present in the `remote`, _but_ - should we log that this is registered at the `/api/preview` route?

`Preview: Server loading API response over HTTP for ${remote}`

**Comment by ecoad** [comment]
Is the cache purging still compatible with Previewing posts?

So `/news/not-yet-published-news` is still purged via `/{purge_key}/news/not-yet-published-news`?

**Comment by jonsherrard** [comment]
@ecoad Good point. Maybe we should always skip the cache if `?tapestry_hash` is present?

File: test/tests/routing.test.js (hunk 1 above)
Comment (thomasdigby) [inline test/tests/routing.test.js]: Tested for both

**Comment by ecoad** [comment]
Skipping cache is a good idea, albeit it is a weakness which may allow a lot of requests to the backend if that link was accidentally shared - which is a fairly likely eventually over time.

File: src/server/handle-preview.js (hunk 2 above)
Comment (thomasdigby) [inline src/server/handle-preview.js]: Yep can do

File: test/tests/routing.test.js (hunk 1 above)
Comment (thomasdigby) [inline test/tests/routing.test.js]: Oh, yes, sorry, my bad

Comment (thomasdigby) [inline test/tests/routing.test.js]: Ah no, sorry, not my bad, it's not present in the URL, only in the custom endpoint

Comment (jonsherrard) [inline test/tests/routing.test.js]: Ah got ya :+1:

File: src/server/handle-preview.js (hunk 2 above)
Comment (thomasdigby) [inline src/server/handle-preview.js]: Let's handle these logs in a separate ticket, we need to create some conventions

**Comment by thomasdigby** [comment]
We're not caching the API response, I'll remove the caching from the HTML as well.

**Pull request merged**